src/shogun/kernel/LinearARDKernel.h

```diff
@@ -4,24 +4,67 @@
  * the Free Software Foundation; either version 3 of the License, or
  * (at your option) any later version.
  *
+ * (W) 2015 Wu Lin
  * (W) 2012 Jacob Walker
  *
  * Adapted from WeightedDegreeRBFKernel.h
  */
 
-#ifndef LINEARARDKERNEL_H_
-#define LINEARARDKERNEL_H_
+#ifndef _LINEARARDKERNEL_H_
+#define _LINEARARDKERNEL_H_
 
 #include <shogun/lib/config.h>
 
 #include <shogun/lib/common.h>
 #include <shogun/kernel/DotKernel.h>
-#include <shogun/features/DenseFeatures.h>
+#include <shogun/mathematics/linalg/linalg.h>
 
 namespace shogun
 {
 
-/** @brief Linear Kernel with Automatic Relevance Detection */
+enum EARDKernelType
+{
+	KT_SCALAR=10,
+	KT_DIAG=20,
+	KT_FULL=30
+};
+
+/** @brief Linear Kernel with Automatic Relevance Detection computed
+ * on CDotFeatures.
+ *
+ * It is computed as
+ *
+ * \f[
+ * k({\bf x},{\bf y})= \frac{||{\bf x}-{\bf y}||}
+ * \f]
+ *
+ * There are three variants based on \f$||\cdot||\f$. 
+ * The default case is
+ * \f$\sum_{i=1}^{p}{{[\lambda \times ({\bf x_i}-{\bf y_i})] }^2}\f$
+ * where \f$\lambda\f$ is a scalar and \f$p\f$ is # of features
+ * To use this case,  please call set_weights(\f$\lambda\f$),
+ * where \f$\lambda\f$ is a 1-by-1 matrix
+ *
+ * The second case is
+ * \f$\sum_{i=1}^{p} {{[\lambda_i \times ({\bf x_i}-{\bf y_i})] }^2}\f$
+ * where \f$\lambda\f$ is a vector (we use \f$\lambda\f$ as a column vector)
+ * and \f$p\f$ is # of features
+ * To use this case,  please call set_weights(\f$\lambda\f$),
+ * where \f$\lambda\f$ is a \f$p\f$-by-1 matrix
+ *
+ * The last case is
+ * \f$({\bf x}-{\bf y})^T \Lambda^T \Lambda ({\bf x}-{\bf y})\f$
+ * where \f$\Lambda\f$ is a \f$d\f$-by-\f$p\f$ matrix,
+ * \f$p\f$ is # of features and \f$d\f$ can be \f$ d \ge p\f$ or \f$ d \le p\f$ 
+ * To use this case,  please call set_weights(\f$\Lambda\f$),
+ * where \f$\Lambda\f$ is a \f$d\f$-by-\f$p\f$ matrix
```

Comment:
as I said below, please have helper methods for this to not confuse users by how to set the weights.
Should be: Scalar, vector, matrix -- as in c++ type